In [94]:
from sklearn.linear_model import LinearRegression #import regression model
import numpy as np
import csv

In [95]:
list1=[]
with open("test_set_C.csv","r") as file:
    fh=csv.reader(file)
    for i in fh:
        list1.append(i)
del list1[0] #import datasets
# NOTE: I already ran this same code for A and C, since the code does not change, I
# am going to provide the final findings in the end

In [96]:
x=np.array(list([float(i[0]),] for i in list1))
y=np.array(list(float(i[1]) for i in list1))
# convert the data into suitable x and y arrays
reg=LinearRegression().fit(x,y)
reg.score(x,y)

0.9999999511116611

In [97]:
list2=[]
with open("test_set_C.csv","r") as file:
    fh=csv.reader(file)
    for i in fh:
        list2.append(i)
del list2[0] #obtain the test dataset

In [98]:
xtest=np.array(list([float(i[0]),] for i in list2))
ytest=np.array(list(float(i[1]) for i in list2))
# make them readable

In [99]:
ypredict=reg.predict(xtest)
ypredict

array([3.96505718e+02, 2.75286546e+03, 2.37736185e+03, 1.26400108e+02,
       1.04910353e+03, 8.83385869e+01, 1.27250603e+03, 1.03344262e+03,
       9.32744779e+02, 1.62920989e+03, 3.35617474e+03, 3.75092941e+03,
       2.67745808e+03, 4.79075019e+02, 7.00517316e+02, 1.13691003e+03,
       2.67598368e+03, 1.89869147e+03, 3.69932960e+03, 3.29077788e+03,
       2.08746382e+03, 3.45139776e+03, 7.76122347e+02, 3.40498505e+03,
       3.14991076e+03, 2.96077438e+03, 3.18298678e+03, 2.99304624e+03,
       3.73409007e+03, 2.45813313e+03, 1.21221412e+03, 4.07232385e+01,
       3.60763386e+03, 2.43163862e+03, 3.04459755e+03, 2.55767241e+03,
       2.59210367e+03, 6.39902476e+02, 2.88732940e+02, 1.45030011e+03,
       3.76488101e+03, 1.16059951e+03, 1.64830499e+03, 3.56112793e+03,
       2.93985667e+03, 3.21563626e+03, 9.33729666e+02, 9.73775317e+02,
       2.68240430e+03, 1.27199921e+03, 2.56066148e+03, 2.94161822e+03,
       3.32614773e+03, 7.14769127e+02, 1.18791634e+03, 1.31266027e+03,
      

In [100]:
error=0
for i in range(len(ytest)):
    error+=(ytest[i]-ypredict[i])**2
error/=len(ytest)
error
## Find mean squared error

0.06227283037374459

The final Mean squared error in the different datasets\
A: 16.140143964067963
\
B: 5.987969894767035e-26
\
C: 0.06227283037374459\
We can see that B performed the best according to the mean squared error, far better than datasets A and C, having basically negligible errors. So it is the real dataset.